src/rule.ts

```diff
@@ -238,7 +238,7 @@ export class Rule {
 							// Defer change notification until the scope of work has completed
 							EventScope$onExit(() => {
 								rule.returnValues.forEach((returnValue) => {
-									(args.entity.changed as Event<Entity, EntityChangeEventArgs>).publish(args.entity, { entity: args.entity, property: returnValue });
+									(args.entity.changed as Event<Entity, EntityChangeEventArgs>).publish(args.entity, { entity: args.entity, property: returnValue, newValue: returnValue.value(args.entity) });
```

Comment:
@tskimmett What problem was this causing? I think this is an odd one because the property hasn't actually changed at this point. I think this is a holdover from old code that I'm not sure is relevant anymore (may be with Vue?).

Reply:
I just changed it because typescript complained... Don't know that it was affecting anything.

Reply:
Was it this line of code that it was complaining about? Or, somewhere else? I'd rather have the `newValue` be optional like `oldValue` (which I thought it was) since because of the way its used its not always possible to provide the value. I don't think its even used though, so I think you could just get rid of `oldValue` and `newValue`. That might actually be a better thing to do anyway.

Reply:
It was that line. I could see `oldValue` being useful. `newValue` can be obtained by just getting the value of that property for the changed entity., so it's redundant.

Reply:
Exactly, and that's what subscribers do currently. I agree `oldValue` could be useful in theory, although I don't think anyone is currently using it. This event is kind of funky and honestly I wouldn't expect anyone to use it externally (other than how its used in VueModel). I think it would be fine to do any of the following (option 2 is my favorite):
1) Satisfy the compiler by making `newValue` optional
2) Remove both `oldValue` and `newValue`
3) Remove just `newValue` and keep `oldValue`
4) Leave it as is and I'll circle back to it at some point (I know this wasn't the point of the changes you were making)

Reply:
I am actually using the entity changed event in application code. Not using old/new value though.

Reply:
Really, what are you using it for?